src/lua/api-gateway/aws/AwsService.lua

```diff
@@ -187,7 +187,12 @@ function _M:getHttpClient()
 end
 
 function _M:getAWSHost()
-    return self.aws_service .. "." .. self.aws_region .. ".amazonaws.com"
+    -- FIXME: the endpoint cannot always be determined mechanically; http://docs.aws.amazon.com/general/latest/gr/rande.html
```

Comment:
I believe an `S3Service.lua` implementation ( similar to [LambdaService.lua](https://github.com/adobe-apiplatform/api-gateway-aws/blob/master/src/lua/api-gateway/aws/lambda/LambdaService.lua) ) may "override" in lua-style this method:

S3Service.lua:
```
local AwsService = require "api-gateway.aws.AwsService"
local cjson = require "cjson"
local error = error

local _M = AwsService:new({ ___super = true })
local super = {
    instance = _M,
    constructor = _M.constructor
}

function _M.new(self, o)
    ngx.log(ngx.DEBUG, "S3Service() o=", tostring(o))
    local o = o or {}
    o.aws_service = "s3"
    -- aws_service_name is used in the X-Amz-Target Header: i.e Kinesis_20131202.ListStreams
    o.aws_service_name = "S3"

    super.constructor(_M, o)

    setmetatable(o, self)
    self.__index = self
    return o
end

function _M:getAWSHost()
    return return self.aws_service .. "-" .. self.aws_region  .. ".amazonaws.com"
end
```

Note `_M:getAWSHost()` method. WDYT ?